src/registries/npm-registry.js

```diff
@@ -218,21 +221,27 @@ export default class NpmRegistry extends Registry {
     }
   }
 
-  getScope(packageName: string): string {
-    return !packageName || packageName[0] !== '@' ? '' : packageName.split(/\/|%2f/)[0];
+  getScope(packageIdent: string): string {
+    // Matches '@scope' in a packageName or pathname, otherwise returns ''
+    if (packageIdent && isScopedPackage(packageIdent)) {
+      const matched = packageIdent.match(/(^@|(?!\/)@).+?((?=%2f)|(?=\/))/);
```

Comment:
I matched on `%2f` because I didn't have the domain knowledge to know for what reason it was escaped in the pathname. If it's not to differentiate them from other forward slashes I could add an extra `.replace('%2f', '/')` call in this line to make the regex more readable.